File: Yunhyunjo/기능개발.cpp
#include <string>
#include <vector>
#include <queue>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    queue <int> q;
    
    for(int i = 0; i < progresses.size(); i++){
        int n = 100 - progresses[i];
        if(n % speeds[i] == 0) n = n / speeds[i];
        else n = n / speeds[i] + 1;
        q.push(n);
    }
    
    int m = q.front();
    q.pop();
    int cnt = 1;
    while(!q.empty()){
        if(m >= q.front()){
            q.pop();
            cnt++;
        }
        else{
            m = q.front();
            q.pop();
            answer.push_back(cnt);
            cnt = 1;
        }
    }
    answer.push_back(cnt);
    return answer;
}

File: gut27/Codingtest/프로그래머스스택,큐/기능개발.py
from collections import deque
# 먼저 들어온 작업이 먼저 나가는 선입 선출 큐 문제
def solution(progresses, speeds):
    answer = []
    progresses = deque(progresses)
    speeds = deque(speeds)
    
    # 모든 작업이 끝날때까지 반복
    while progresses:
        # 개발
        for i in range(len(progresses)):
            if progresses[i] >= 100:
                continue
            progresses[i] += speeds[i]
        
        # 배포
        count = 0
        while progresses:
            if progresses[0] >= 100:
                count+=1
                speeds.popleft()
                progresses.popleft()
            else:
                break
        
        if count > 0:
            answer.append(count)
    return answer
